adobe/spectrum-css
# feat(coachmark)!: migrate tokens

## Description




Migrates tokens and implements new styling for coach mark 
- Splits the coach mark indicator and coach mark popover into 2 separate components, `CoachIndicator` and `CoachMark`
- is "responsive" and displays different button and menu styling for Large scale (mobile) 
- Updates stories for both components

[Jira Ticket - CSS 193](https://jira.corp.adobe.com/browse/CSS-193) 

[XD File](https://adobe.sharepoint.com/sites/Spectrum/Shared%20Documents/Forms/AllItems.aspx?csf=1&web=1&e=CS3eWZ&cid=d45bcec1%2D06b4%2D401f%2D8f1e%2D6b674ae78a85&FolderCTID=0x0120004AB1576A94E3914C949954710369C339&id=%2Fsites%2FSpectrum%2FShared%20Documents%2FSpectrum%20Design%2FToken%20specs%2FCoach%20Mark%2Exd&parent=%2Fsites%2FSpectrum%2FShared%20Documents%2FSpectrum%20Design%2FToken%20specs)

## How and where has this been tested?

- **How this was tested:** 
- **Browser(s) and OS(s) this was tested with:** 
Browser(s) and OS(s) this was tested with:
Chrome Version 113.0.5672.63 on macOS
Safari 16.4 on macOS
Firefox 112.0.2 on macOS

### Regression testing

#### Test Outline @jawinn
1. Open the [docs site](https://pr-1992--spectrum-css.netlify.app/coachmark) for the CoachMark component.
- [x] component should display mobile styling when scale is Large
- [x] styling should match XD files 

2. Open the [storybook site](https://pr-1992--spectrum-css.netlify.app/preview/?path=/story/components-coach-mark--default) for the CoachMark component
- [x] controls should all function as expected 
- [x] styling should match docs site 


3. Open the [docs site](https://pr-1992--spectrum-css.netlify.app/coachindicator) for the CoachIndicator component.
- [x] Styling should match the live docs site 

4. Open the [storybook site](https://pr-1992--spectrum-css.netlify.app/preview/?path=/docs/components-coach-indicator--docs) for the CoachIndicator component
- [x] controls should all function as expected _@jawinn : except as noted in comment about reduceMotion_
- [x] styling should match docs site

Validate: @jawinn 

1. A legacy documentation page (such as [accordion](https://pr-###--spectrum-css.netlify.app/accordion.html)), including:


2. A migrated documentation page (such as [action group](https://pr-###--spectrum-css.netlify.app/actiongroup.html)), including:


## Screenshots

### Desktop (medium)
![Screenshot 2023-08-24 at 11 06 59 AM](https://github.com/adobe/spectrum-css/assets/63808889/bebb6066-47b3-4106-9dbb-cb1dbfc509e5)
![Screenshot 2023-08-24 at 11 06 48 AM](https://github.com/adobe/spectrum-css/assets/63808889/0cbfba39-9f0d-4a55-bc64-2b726ccf516f)

### Mobile (Large)
![Screenshot 2023-08-24 at 11 07 57 AM](https://github.com/adobe/spectrum-css/assets/63808889/d0ad0c4d-fd1f-4c17-8e07-cdd476462d20)
![Screenshot 2023-08-24 at 11 08 06 AM](https://github.com/adobe/spectrum-css/assets/63808889/d8bbfef2-a221-4ac2-bc81-008a3cb55b7e)



## To-do list



- [x] I have updated any relevant storybook stories and templates.
- [x] If my change(s) include visual change(s), a designer has reviewed and approved those changes.
